### PR TITLE
eastl: add version 3.21.12, add patch descriptions

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **eastl/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/eastl/all/conandata.yml
Comment (RubenRBS) [inline recipes/eastl/all/conandata.yml:34]: [The linked PR](https://github.com/electronicarts/EASTL/pull/513/files) has more contents in the patch than what is in this file, should we use the whole diff instead of just the one that was making it fail to compile?

**Review by RubenRBS — COMMENTED** [review]
Thanks! Minor comment about the patch, but looks great!

File: recipes/eastl/all/conandata.yml
Comment (toge) [reply recipes/eastl/all/conandata.yml:34]: @RubenRBS 
I think that no further modifications are necessary.
This is because the other source codes (array.h, span.h) have been modified since 3.21.12.
https://github.com/electronicarts/EASTL/blob/3.21.12/include/EASTL/array.h
https://github.com/electronicarts/EASTL/blob/3.21.12/include/EASTL/span.h

Comment (RubenRBS) [reply recipes/eastl/all/conandata.yml:34]: Oki, thanks for the clarification :)

Let's wait until we get a response in https://github.com/electronicarts/EASTL/issues/512#issuecomment-1623476892 to clarify the better way forward

**Review by RubenRBS — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]
LGTM, including the backported patch as advised by EASTL maintainers: https://github.com/electronicarts/EASTL/issues/512#issuecomment-1623476892

**Pull request merged**
